File: dotnet/v2.6/Placements/DownloadPlacementTags.cs
/*
 * Copyright 2015 Google Inc
 *
 * Licensed under the Apache License, Version 2.0(the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

using System;
using Google.Apis.Dfareporting.v2_6;
using Google.Apis.Dfareporting.v2_6.Data;

namespace DfaReporting.Samples {
  /// <summary>
  /// This example downloads HTML Tags for a given campaign and placement ID.
  /// To create campaigns, run CreateCampaign.cs. To create placements, run
  /// CreatePlacement.cs.
  /// </summary>
  class DownloadPlacementTags : SampleBase {
    /// <summary>
    /// Returns a description about the code example.
    /// </summary>
    public override string Description {
      get {
        return "This example downloads HTML Tags for a given campaign and" +
            " placement ID. To create campaigns, run CreateCampaign.cs. To" +
            " create placements, run CreatePlacement.cs.\n";
      }
    }

    /// <summary>
    /// Main method, to run this code example as a standalone application.
    /// </summary>
    /// <param name="args">The command line arguments.</param>
    public static void Main(string[] args) {
      SampleBase codeExample = new DownloadPlacementTags();
      Console.WriteLine(codeExample.Description);
      codeExample.Run(DfaReportingFactory.getInstance());
    }

    /// <summary>
    /// Run the code example.
    /// </summary>
    /// <param name="service">An initialized Dfa Reporting service object
    /// </param>
    public override void Run(DfareportingService service) {
      long campaignId = long.Parse(_T("ENTER_CAMPAIGN_ID_HERE"));
      long placementId = long.Parse(_T("ENTER_PLACEMENT_ID_HERE"));
      long profileId = long.Parse(_T("INSERT_USER_PROFILE_ID_HERE"));

      // Generate the placement activity tags.
      PlacementsResource.GeneratetagsRequest request =
          service.Placements.Generatetags(profileId);
      request.CampaignId = campaignId;
      request.TagFormats =
          PlacementsResource.GeneratetagsRequest.TagFormatsEnum.PLACEMENTTAGSTANDARD;
      request.PlacementIds = placementId.ToString();

      PlacementsGenerateTagsResponse response = request.Execute();

      // Display the placement activity tags.
      foreach (PlacementTag tag in response.PlacementTags) {
        foreach (TagData tagData in tag.TagDatas) {
          Console.WriteLine("{0}:\n", tagData.Format);

          if (tagData.ImpressionTag != null) {
            Console.WriteLine(tagData.ImpressionTag);
          }

          if (tagData.ClickTag != null) {
            Console.WriteLine(tagData.ClickTag);
          }

          Console.WriteLine();
        }
      }
    }
  }
}

File: ruby/v3.2/placements/download_placement_tags.rb
#!/usr/bin/env ruby

#
# Copyright:: Copyright 2016, Google Inc. All Rights Reserved.
#
# License:: Licensed under the Apache License, Version 2.0 (the "License");
#           you may not use this file except in compliance with the License.
#           You may obtain a copy of the License at
#
#           http://www.apache.org/licenses/LICENSE-2.0
#
#           Unless required by applicable law or agreed to in writing, software
#           distributed under the License is distributed on an "AS IS" BASIS,
#           WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
#           implied.
#           See the License for the specific language governing permissions and
#           limitations under the License.
#
# This example downloads HTML Tags for a given campaign and placement ID.
#
# To create campaigns, run create_campaign.rb. To create placements, run
# create_placement.rb.

require_relative '../dfareporting_utils'

def download_placement_tags(profile_id, campaign_id, placement_id)
  # Authenticate and initialize API service.
  service = DfareportingUtils.get_service

  # Construct the request.
  result = service.generate_placement_tags(profile_id,
    campaign_id: campaign_id,
    placement_ids: [placement_id])

  result.placement_tags.each do |tag|
    tag.tag_datas.each do |data|
      puts format("%s - %s\n\n", tag.placement_id, data.format)
      puts format("%s\n\n", data.impression_tag) unless data.impression_tag.nil?
      puts format("%s\n\n", data.click_tag) unless data.click_tag.nil?
    end
  end
end

if $PROGRAM_NAME == __FILE__
  # Retrieve command line arguments.
  args = DfareportingUtils.get_arguments(ARGV, :profile_id, :campaign_id,
    :placement_id)

  download_placement_tags(args[:profile_id], args[:campaign_id],
    args[:placement_id])
end
